book/jvm-rebuilds-service.md

```diff
@@ -0,0 +1,167 @@
+# Java Virtual Machine Build Service
+
+## Overview
+
+The Java Virtual Machine Build Service (JBS, or JVM Build Service) is a controller that will rebuild Java and other JVM Language based dependencies from source.
+
+Java uses a binary distribution model, so the only way to ensure an application is completely built from source is to rebuild all its component libraries from source themselves. Unfortunately due to the nature of the Java ecosystem this is not an easy process and would normally require a large amount of manual toil.
```

Comment:
Binary distribution as opposed to source distribution? I assume that this just means that the binaries are the only form of possible distribution and that that is enabled because of the abstraction provided by the JVM.

If this is accurate, you don't need to include it all (or any of it). I just wanted to make sure that I understood what you were saying and not saying.

Reply:
It's not so much that binaries are the only form of possible distribution, in theory you could write a source based build tool, its just that in practice its what everyone uses.